httomo/runner/loader.py

```diff
@@ -78,6 +78,102 @@ class UserDefinedAngles(NamedTuple):
 
 AnglesConfig: TypeAlias = Union[RawAngles, UserDefinedAngles]
 
+class PreviewDimConfig(NamedTuple):
+    start: int
+    stop: int
+
+class PreviewConfig(NamedTuple):
+    angles: PreviewDimConfig
+    detector_y: PreviewDimConfig
+    detector_x: PreviewDimConfig
+
+class Preview:
+    def __init__(
+        self,
+        preview_config: PreviewConfig,
+        dataset: h5py.Dataset,
+        image_key: Optional[h5py.Dataset],
+    ) -> None:
+        self.config = preview_config
+        self._dataset = dataset
+        self._image_key = image_key
+        self._check_within_data_bounds()
+        self._data_indices: Optional[List[int]] = None
+        self._global_shape: Optional[Tuple[int, int, int]] = None
+
+    def _check_within_data_bounds(self) -> None:
+        shape = self._dataset.shape
+        for i, field in enumerate(self.config._fields):
+            self._check_dimension(
+                name=field,
+                config=getattr(self.config, field),
+                length=shape[i],
+            )
+
+    def _check_dimension(
+        self,
+        name: str,
+        config: PreviewDimConfig,
+        length: int,
+    ) -> None:
+        if config.stop > length:
+            raise ValueError(
+                f"Preview indices in {name} dim exceed bounds of data: "
+                f"start={config.start}, " f"stop={config.stop}"
+            )
+
+        if config.start >= config.stop:
+            raise ValueError(
+                f"Start index in preview indices for {name} dim is >= stop index: "
```

Comment:
Suggestions like this are really useful, when in the weeds of development it can be hard to come up with an error message that would make sense to users. I like your suggestion, I'll add that change!